sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -535,6 +542,77 @@ pub fn tuf_prefix(
     Ok(())
 }
 
+/// The `aws_config` helper is used to create an AWS config file
+/// with `use_fips_endpoint` value set based on if the variant is FIPS enabled.
+///
+/// # Fallback
+///
+/// If this helper runs and `settings.aws.config` is set,
+/// the helper will return early, leaving the existing content intact.
+///
+/// # Example
+///
+/// The AWS config value is generated via
+/// `{{ aws_config settings.aws.config settings.aws.profile }}`
+///
+/// This would result in something like:
+/// ```toml
+/// [default]
+/// use_fips_endpoint=false
+/// ```
+///
+/// The helper will then base64 encode this content and set as `settings.aws.config`.
+pub fn aws_config(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting aws-config-fips-endpoint helper");
+    let template_name = template_name(renderctx);
+
+    check_param_count(helper, template_name, 2)?;
+
+    let aws_config = get_param(helper, 0)?;
+    if let Value::Null = aws_config {
+        // get the profile parameter, which is probably given by the template value
+        // settings.aws.profile. Regardless, it is expected to be a string.
+        let aws_profile = get_param(helper, 1)?;
+        let aws_profile = aws_profile
```

Comment:
We should not assume `aws_profile` is defined; it could be null if the downstream build doesn't define a default value for it.

Reply:
Does that also then apply to some of the other helpers we have, such as `ecr-prefix` (e.g. if a downstream has no default region): https://github.com/bottlerocket-os/bottlerocket-core-kit/blob/a10182f974abd19de803bb7a324b5a0bdeab4ced/sources/api/schnauzer/src/helpers/mod.rs#L467-L473

My understanding is that if this isn't defined, in both this new helper and in `ecr-prefix`, the helper will fail. Would it be preffered to just silently return and do nothing?

Reply:
> Does that also then apply to some of the other helpers we have, such as ecr-prefix  ...?

Likely yes - this is an artifact of the in-tree world, where we "knew" that all variants would have a default value because we could review or update the defaults at the same time we added these helpers.

Reply:
Sounds good. I can backlog an issue for that. As for this PR, I've removed the dependency on `settings.aws.profile` for this helper given our other comment thread discussing only setting this config for the `default` profile